packages/react-query/src/useBaseQuery.ts

```
@@ -50,9 +51,17 @@ export function useBaseQuery<
   }
 
   const isRestoring = useIsRestoring()
+  const hydratingQueriesRef = useIsHydrating()
   const errorResetBoundary = useQueryErrorResetBoundary()
   const client = useQueryClient(queryClient)
   const defaultedOptions = client.defaultQueryOptions(options)
+
+  // Check if this query is pending hydration (using mutable ref)
+  // eslint-disable-next-line react-hooks/refs
```

ESLint

[error] Definition for rule 'react-hooks/refs' was not found.

(react-hooks/refs)